internal/server/server.go

```diff
@@ -100,90 +112,160 @@ func (s *Service) Run() error {
-	s.registerComputeTools(s.azClient)
+	// Optional Kubernetes Components (based on configuration)
+	s.registerOptionalKubernetesComponents()
 
-	// TODO: Add other resource categories in the future:
+	log.Println("Kubernetes Components registered successfully")
 }
 
-// registerNetworkTools registers the network resources tool
-func (s *Service) registerNetworkTools(azClient *azureclient.AzureClient) {
-	log.Println("Registering Network tool...")
+// registerKubectlComponent registers core kubectl commands based on access level
+func (s *Service) registerKubectlComponent() {
+	log.Println("Registering Core Kubernetes Component (kubectl)")
 
-	// Register network resources tool
-	log.Println("Registering network tool: az_network_resources")
-	networkTool := network.RegisterAzNetworkResources()
-	s.mcpServer.AddTool(networkTool, tools.CreateResourceHandler(network.GetAzNetworkResourcesHandler(azClient, s.cfg), s.cfg))
+	// Get kubectl tools filtered by access level
+	kubectlTools := kubectl.RegisterKubectlTools(s.cfg.AccessLevel)
+
+	// Create a kubectl executor
+	kubectlExecutor := kubectl.NewKubectlToolExecutor()
 
+	// Convert aks-mcp config to k8s config
+	k8sCfg := k8s.ConvertConfig(s.cfg)
+
+	// Register each kubectl tool
+	for _, tool := range kubectlTools {
+		log.Printf("Registering kubectl tool: %s", tool.Name)
+		// Create a handler that injects the tool name into params
+		handler := k8stools.CreateToolHandlerWithName(kubectlExecutor, k8sCfg, tool.Name)
+		s.mcpServer.AddTool(tool, handler)
+	}
 }
 
-// registerDetectorTools registers all detector-related Azure resource tools
-func (s *Service) registerDetectorTools(azClient *azureclient.AzureClient) {
-	log.Println("Registering Detector tools...")
+// registerOptionalKubernetesComponents registers optional Kubernetes tools based on configuration
+func (s *Service) registerOptionalKubernetesComponents() {
+	log.Println("Registering Optional Kubernetes Components")
 
-	// Register list detectors tool
-	log.Println("Registering detector tool: list_detectors")
-	listTool := detectors.RegisterListDetectorsTool()
-	s.mcpServer.AddTool(listTool, tools.CreateResourceHandler(detectors.GetListDetectorsHandler(azClient, s.cfg), s.cfg))
+	// Register helm if enabled
+	s.registerHelmComponent()
 
-	// Register run detector tool
-	log.Println("Registering detector tool: run_detector")
-	runTool := detectors.RegisterRunDetectorTool()
-	s.mcpServer.AddTool(runTool, tools.CreateResourceHandler(detectors.GetRunDetectorHandler(azClient, s.cfg), s.cfg))
+	// Register cilium if enabled
+	s.registerCiliumComponent()
 
-	// Register run detectors by category tool
-	log.Println("Registering detector tool: run_detectors_by_category")
-	categoryTool := detectors.RegisterRunDetectorsByCategoryTool()
-	s.mcpServer.AddTool(categoryTool, tools.CreateResourceHandler(detectors.GetRunDetectorsByCategoryHandler(azClient, s.cfg), s.cfg))
+	// Register Inspektor Gadget tools for observability
```

Comment:
Fyi...I am working on enabling the [IG tool by default](https://github.com/Azure/aks-mcp/pull/115), so we might need to rework things a little depending on whichever goes in first!

Reply:
@mqasimsarfraz I just rebased the branch after your change in master, please help to check if I missed anything.